src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8922,10 +8925,28 @@ private ExpressionSyntax ParseSubExpressionCore(Precedence precedence)
             if (IsExpectedPrefixUnaryOperator(tk))
             {
                 opKind = SyntaxFacts.GetPrefixUnaryExpression(tk);
-                newPrecedence = GetPrecedence(opKind);
                 var opToken = this.EatToken();
-                var operand = this.ParseSubExpression(newPrecedence);
-                leftOperand = _syntaxFactory.PrefixUnaryExpression(opKind, opToken, operand);
+
+                if (opToken.Kind == SyntaxKind.DotDotToken)
+                {
+                    // Operator ".." here can either be a prefix unary operator or a stand alone empty range:
+                    Debug.Assert(opKind == SyntaxKind.RangeExpression);
+
+                    if (IsPossibleExpression(allowBinaryExpressions: false, allowAssignmentExpressions: false))
+                    {
+                        newPrecedence = GetPrecedence(opKind);
+                        leftOperand = _syntaxFactory.RangeExpression(leftOperand: null, opToken, rightOperand: this.ParseSubExpression(newPrecedence));
+                    }
+                    else
+                    {
+                        leftOperand = _syntaxFactory.RangeExpression(leftOperand: null, opToken, rightOperand: null);
+                    }
```

Comment:
Consider parsing `rightOperand` f the range expression inside the `if/else` and call `_syntaxFactory.RangeExpression()` after the `if/else`.